streams/src/test/java/org/apache/kafka/streams/processor/internals/tasks/DefaultTaskManagerTest.java

```diff
@@ -137,6 +137,8 @@ public void shouldBlockOnAwait() throws InterruptedException {
         assertFalse(awaitingRunnable.awaitDone.await(100, TimeUnit.MILLISECONDS));
 
         awaitingRunnable.shutdown();
+        awaitingThread.join();
+        assertFalse(awaitingThread.isAlive());
```

Comment:
Not sure if we need to add this check, as `join()` would block forever anyway (so it seems we either timeout, or know it will return `true` and the check is redundant).

Or as an alternative, keep this check, but add a max wait time to `join()`.

Thoughts?

Reply:
Yeah, I think I'd just `join` the thread